Paintroid/src/androidTest/java/org/catrobat/paintroid/test/espresso/tools/TransformToolIntegrationTest.kt

```diff
@@ -0,0 +1,1367 @@
+        onToolBarView()
+            .performSelectTool(ToolType.TRANSFORM)
+        var seekBar =
+            mainActivity.findViewById<SeekBar>(R.id.pocketpaint_transform_resize_seekbar)
+        var progress = seekBar.progress
+        onTransformToolOptionsView()
+            .checkPercentageTextMatches(progress)
+        onTransformToolOptionsView()
+            .moveSliderTo(1)
+        onToolBarView()
+            .performSelectTool(ToolType.BRUSH)
+        onToolBarView()
+            .performSelectTool(ToolType.TRANSFORM)
+        seekBar =
+            launchActivityRule.activity.findViewById(R.id.pocketpaint_transform_resize_seekbar)
+        progress = seekBar.progress
+        onTransformToolOptionsView()
+            .checkPercentageTextMatches(progress)
+        onTransformToolOptionsView()
+            .performApplyResize()
+        onTransformToolOptionsView()
+            .moveSliderTo(50)
+        progress = seekBar.progress
+        onTransformToolOptionsView()
+            .checkPercentageTextMatches(progress)
+        onToolBarView()
+            .performSelectTool(ToolType.BRUSH)
+        onToolBarView()
+            .performSelectTool(ToolType.TRANSFORM)
+        onTransformToolOptionsView()
+            .performApplyResize()
+        seekBar =
+            launchActivityRule.activity.findViewById(R.id.pocketpaint_transform_resize_seekbar)
+        progress = seekBar.progress
+        onTransformToolOptionsView()
+            .checkPercentageTextMatches(progress)
+    }
+
+    @Test
+    fun testTransformToolDoesNotResetPerspectiveScale() {
+        val scale = 2.0f
+        perspective.scale = scale
+        perspective.setSurfaceTranslationX(50f)
+        perspective.setSurfaceTranslationY(200f)
+        mainActivity.refreshDrawingSurface()
+        onToolBarView()
+            .performSelectTool(ToolType.TRANSFORM)
+        assertEquals(scale, perspective.scale, 0.0001f)
+    }
+
+    companion object {
+        private fun drawPlus(bitmap: Bitmap, lineLength: Int) {
+            val horizontalStartX = bitmap.width / 4
+            val horizontalStartY = bitmap.height / 2
+            val verticalStartX = bitmap.width / 2
+            val verticalStartY = bitmap.height / 2 - lineLength / 2
+            val pixelsColorArray = IntArray(10 * lineLength)
+            for (indexColorArray in pixelsColorArray.indices) {
+                pixelsColorArray[indexColorArray] = Color.BLACK
+            }
+            bitmap.setPixels(
+                pixelsColorArray,
+                0,
+                lineLength,
+                horizontalStartX,
+                horizontalStartY,
+                lineLength,
+                10
+            )
+            bitmap.setPixels(
+                pixelsColorArray,
+                0,
+                10,
+                verticalStartX,
+                verticalStartY,
+                10,
+                lineLength
+            )
+        }
+    }
+}
```

Comment:
This test looks worse in kotlin. Please revert this change and make an extra ticket in which improvements to the test are then applied.

Reply:
The test would fail otherwise 😅

Reply:
Can't think of a reason why it would fail. :shrug: 
Please make the necessary change in java, the kotlin test code is unreadable.

Reply:
runBlocking won't be possible in Java

Reply:
Hmm, that's unfortunate... Can you extract the affected test cases into a file? (Since this test file is anyway too big)